tests/algorithm/scripts/send_test_records.py

```diff
@@ -0,0 +1,66 @@
+import csv
+
+import requests
+from helpers import dict_to_pii
+
+
+def send_test_records(test_csv, algorithm_name, api_url):
+    output_data = []
+
+    print("Sending test records to the API...")
+
+    # Read the CSV file using the csv module
+    with open(test_csv, mode='r', newline='', encoding='utf-8') as file:
+        reader = csv.DictReader(file)
+
+        for row in reader:
+            # Replace None and "NaN" values with an empty string
+            record_data = {k: ("" if v in [None, "NaN"] else v) for k, v in row.items()}
+
+            # Get info from row
+            match_info = {
+                "test_case_number": record_data['Test Case #'],
+                "match_id": record_data['Match Id'],
+                "should_match": record_data['Expected Result'],
+            }
+
+            pii_record = dict_to_pii(record_data)
+            
+            response = send_record(pii_record, algorithm_name, api_url)
+
+            if response:
+                output_row = {
+                    "Test Case #": match_info['test_case_number'],
+                    "Expected Result": match_info['should_match'],
+                    "Match Result": response['prediction'],
+                    "Error": ""
+                }
+            else:
+                output_row = {
+                    "Test Case #": match_info['test_case_number'],
+                    "Expected Result": match_info['should_match'],
+                    "Match Result": "Failed",
```

Comment:
What would cause this `"Failed"` result?

Would we want to error out in the testing framework somewhere earlier on? Not sure how long this kind of testing would take to run, but I'm imaging the case when it takes a while only to return many `"Failed to link record"` results due to a formatting error or something easily fixed; the user would only become aware of that after the testing finished running.

Reply:
The failed result happens when a record is sent to be linked via the `/link` endpoint but returns some sort of error. Could be a couple of causes but the most likely cause is just bad data. Invalid formatting or too many characters are common ones. I could print more information related to the error in the output file or in the terminal. My only thought with erroring out earlier is that the test may almost be complete and then suddenly get stopped. My thought is that if the user sees and error in the terminal then they can stop the test if they want but otherwise the program fails that test and moves on.

Reply:
I think printing more information related to the error would be helpful, but I think this is a good discussion that brings up the question of: is the purpose of this testing framework to test the functionality of Record Linker (e.g., do not pass invalid patient records), or the match accuracy of the algorithm it uses?

I believe the latter, although allowing `"Failed to link record"` results as described above would be testing more of the functionality of Record Linker, I think. I think the goal of this kind of match accuracy testing is to, once the tests are complete, have an idea about how accurate the current algorithm is (not whether there were data formatting issues). Calculating match accuracy metrics like Sensitivity, Specificity, etc. is out of scope for this PR (I believe I'll be working on that as a fast-follow afterwards), but doing so will require "clean" prediction results (`"match"`, `"no_match"`, `"possible_match"`). It will be difficult to interpret and calculate these metrics when we have results like `"Failed to link result"`. For example, would we just exclude those results in the calculations? If so, we may report flawed accuracy metrics (what if the patient records with `"Failed to link result"` would have been accurately linked, but the user just had a data formatting issue?).

Interested in what you both think regarding if we should standardize the`"Expected Result"` and `"Match Result"` values here. IMO, it would be better to standardize these values in the testing framework rather than when calculating Sensitivity, Specificity, etc. CC: @ericbuckley

Reply:
Yeah, I agree that the goal is not to fix errors in the user's data but rather to analyze the results of the algorithm configuration on the user's data with their test cases. I'm thinking the simple "Failed to link record" or something like that would be all we need right now since we aren't calculating any metrics yet as you said. And I know NBS even said they had some test cases with intentionally bad data that are expected to fail. 

Right now the result file shows "failed to link record" as well as a short error description. Do we think that's too much or too little info for the users at this point? Also if either of y'all have word smithing on that then feel free to let me know. this was just the first thing I thought of.

@alhayward @ericbuckley

Reply:
@alhayward, Cason and I spoke about this after standup today and decided that we should do the following.

a) Distinguish between server and user errors, the latter can be returned to the user in the match results, but the former should cause the program to halt immediately (since its not expected).
b) User errors ([a 422](https://cdcgov.github.io/RecordLinker/api-docs.html#tag/link/operation/link_piirecord_link_post)), will show the problem in the detail attribute of the payload, we can extract this information and put it into the Error column for the user to see.